protocol/0078-NWLI-network_wide_limits.md

```diff
@@ -0,0 +1,80 @@
+
+## Number of markets
+
+Introduce a new network parameter `limits.markets.max` controlling the maximum number of markets allowed within a network.
+
+Each market in a `Proposed` [state](./0043-MKTL-market_lifecycle.md) should increment the **total market count** by 1. If a market reaches a `Rejected`, `Cancelled`, `Closed` or `Settled` state the total market count should be decremented by 1.
+As soon as the total market count reaches the value specified by `limits.markets.max` parmeter no other further market proposals should be accepted. If the total market count drops below `limits.markets.max` the network should start accepting market proposals again.
+
+### Change of network parameter
+
+If the network parameter gets increased via a governance vote no further actions are needed.
+If it gets decreased below the current total market count then all existing markets continue functioning in their respective states, but no further proposals are accepted until the count drops below the new limit.
+
+## Parties in a market
+
+Introduce a new network parameter `limits.markets.maxParties` controlling the maximum number of parties allowed within any given market.
+
+A party gets counted towards the limit if it has either open orders or open positions in the market. Once party has no open orders and no open positions it gets removed from a **total party count** within the market. Once the limit gets reached the market accepts no further orders.
+Once the total party count drops below the limit market accepts orders again (provided its [state](./0043-MKTL-market_lifecycle.md) allows that).
+
+The limit does not apply to liquidity providers.
+
+### Change of network parameter
+
+If the network parameter gets increased via a governance vote no further actions are needed.
+If it gets decreased below the current total party count in any given market then all parties in that market remain unaffected (their positions and orders remain open), but no new parties are allowed to participate in the market until the total party count drops below the new limit.
+
+## Limit orders on a book
+
+Introduce a new network parameter `limits.markets.maxLimitOrders` controlling the maximum number of limit orders that can rest on a book in any market. If the limit is reached limit orders can still be submitted, but:
+
+* if the order is aggressive (results in a trade) the trade proceeds as normal,
+* if the order is passive (it doesn't match on entry) and would get added to the order book it gets rejected.
+
+Pegged orders and [liquidity provision orders](./0038-OLIQ-liquidity_provision_order_type.md) do not get counted towards the limit.
```

Comment:
The consequence of that is that if the maximum number of orders gets reached in a market then LPs can only exercise their strategy via the automated order shapes. This may not be ideal, but allowing exclusions for LPs would complicate the design and also open up the door for malicious party entering the market as an LP to circumvent the limits, so I'd leave it as is specified here for the first implementation and change it only if we can clear requests from LPs that it impedes their ability to exercise their strategy.